src/Features/ExternalAccess/Copilot/Internal/Completion/CSharpContextProviderService.cs

```diff
@@ -6,80 +6,41 @@
 using System.Collections.Generic;
 using System.Collections.Immutable;
 using System.Composition;
-using System.Linq;
-using System.Runtime.CompilerServices;
 using System.Threading;
-using System.Threading.Tasks;
-using Microsoft.CodeAnalysis;
-using Microsoft.CodeAnalysis.ErrorReporting;
 using Microsoft.CodeAnalysis.ExternalAccess.Copilot.Completion;
 using Microsoft.CodeAnalysis.Host.Mef;
-using Microsoft.VisualStudio.Threading;
+using Microsoft.CodeAnalysis.Shared.Utilities;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.Copilot.Internal.Completion;
 
-[Shared]
-[Export(typeof(ICSharpCopilotContextProviderService))]
-internal sealed class CSharpContextProviderService : ICSharpCopilotContextProviderService
+[Export(typeof(ICSharpCopilotContextProviderService)), Shared]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal sealed class CSharpContextProviderService([ImportMany] IEnumerable<IContextProvider> providers)
+    : ICSharpCopilotContextProviderService
 {
-    // Exposed for testing
```

Comment:
note: i couldn't find this actually read anywhere.  if it is used for testing, it should be exposed through a GetTestAccessor() method.

Reply:
oops, I forgot to move the tests over.